File: javascript/leetcode_7.js
/**
 * @param {number} x
 * @return {number}
 */


var reverse = function(x) {

    if (x <= -2147483648 || x == 0){
        return 0;
    }

    num = Math.abs(x);
    let reverseInteger = 0;

    while (true){
        let digit = num%10;
        reverseInteger = reverseInteger*10 + digit;
        num = Math.floor(num/10);
        if (num == 0){
            break
        }
    }

    if (reverseInteger <= 2147483647 ){
        return (x/(Math.abs(x)))*reverseInteger
    }else{
        return 0;        
    }
        
};

console.log(reverse(123));
console.log(reverse(1563847412));

File: python/leetcode_7.py

@profile
def reverse( x: int) -> int:
    num = abs(x)
    list_of_digits = []
    reverse_string = ''
    while True:
        #q,rem = divmod(num,10)
        digit = num%10
        reverse_string += digit
        num = num // 10
        if num == 0:
            break
    reverse_integer = int(reverse_string)
    if (reverse_integer >= -2147483648 and reverse_integer <= 2147483647 ):
        if x == 0:
            return reverse
        else:
            return (x//(abs(x)))*reverse_integer
    else:
        return 0
            

if __name__ == '__main__':
    print(reverse(214748364))